autosklearn/metalearning/metalearning/meta_base.py

```diff
@@ -55,8 +58,13 @@ def __init__(self, configuration_space, aslib_directory, logger):
         for algorithm_id in self.configurations:
             configuration = self.configurations[algorithm_id]
             try:
+                for key in self.default_configuration_space_dict.keys():
+                    if key not in configuration:
+                        configuration[key] = self.default_configuration_space_dict[key]
                 configurations[str(algorithm_id)] = Configuration(
-                    configuration_space, values=configuration
+                    configuration_space,
+                    values=configuration,
+                    allow_inactive_with_values=True,
```

Comment:
I'm not sure this will work but I could be wrong.
Basically I'm concerned that the `self.default_configuration_space_dict` is just a single configuration like an `svm` or something and then when you iterate through the read in configurations, and fill in those values, you may end up putting in values that shouldn't be there, e.g. you load in a `LinearRegressor` config and then give it `svm` hyperparamters.

Reply:
Yes i see added ` allow_inactive_with_values=True` this is allowing to provided more hps as there are in the search space and i think they are ignored if they dont match. Also if `LinearRegressor` and `SVMRegressor` share a hyperparameter name it is picked from `configuration` and not from the `default_configuration`. But i also not entirly sure about the  ` allow_inactive_with_values=True`  behavior.

Reply:
Okies, good to know and thanks for clarifying :)